File: java/src/main/java/webmis/util/Hmac.java
package webmis.util;

import java.nio.charset.StandardCharsets;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;

import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;

import org.springframework.util.DigestUtils;

/* 加密 */
public class Hmac {

  /* Md5 */
  public static String Md5(String data) {
    return DigestUtils.md5DigestAsHex(data.getBytes());
  }

  /* Sha256 */
  public static byte[] Sha256(String data, String key) {
    try {
      byte[] byteKey = key.getBytes(StandardCharsets.UTF_8);
      Mac hmac = Mac.getInstance("HmacSHA256");
      SecretKeySpec keySpec = new SecretKeySpec(byteKey, "HmacSHA256");
      hmac.init(keySpec);
      return hmac.doFinal(data.getBytes(StandardCharsets.UTF_8));
    } catch (NoSuchAlgorithmException | InvalidKeyException e) {
      return null;
    }
  }
  
}

File: python/util/hmac.py
import hashlib
import hmac

# 加密
class Hmac:

  # Md5
  def Md5(data: str):
    return hashlib.md5(data.encode(encoding='utf-8')).hexdigest()

  # Sha256
  def Sha256(data: str, key: str):
    return hmac.new(str.encode(key), str.encode(data), hashlib.sha256).digest()
